src/System.IO.FileSystem/tests/FileInfo/GetSetTimes.cs

```diff
@@ -36,6 +36,24 @@ public FileInfo GetNonZeroMilliSec()
             return fileinfo;
         }
 
+        public FileInfo GetNonZeroNanoSeconds()
+        {
+            FileInfo fileinfo = new FileInfo(GetTestFilePath());
+            for (int i = 0; i < 5; i++)
+            {
+                fileinfo.Create().Dispose();
+                if (fileinfo.LastWriteTime.Ticks % 10 != 0)
+                    break;
+
+                // This case should only happen 1/10 times, unless the OS/Filesystem does
+                // not support nanosecond granularity.
+
+                // If it's 1/10, or low granularity, this may help:
+                Thread.Sleep(123);
+            }
```

Comment:
If it ends up failing all 5 times, won't we return a file that has zero for nano seconds?  And then subsequent tests might fail and it won't be obvious that they failed because we couldn't create the file correctly.  We should Assert before returning.